Models/Board.cs

```diff
@@ -1,33 +1,40 @@
 ﻿using Chess_Backend.Models.Pieces;
 using Chess_Backend.Models.Positions;
-using System;
-using System.Collections.Generic;
-using System.Text;
-
 namespace Chess_Backend.Models
 {
     public class Board : IBoard
     {
         public List<Piece> Pieces { get; private set; }
-        public Dictionary<Tile, Piece> Position { get; private set; }
+        public Dictionary<Tile, Piece> Positions { get; private set; }
+        public Color currentTurnColor { get; private set; }
 
-        public Board(List<Piece> pieces)
+        public Board()
+        {
+            Pieces = new List<Piece>();
+            Positions = new Dictionary<Tile, Piece>();
+            MapPiecesToDictionary();
+
+        }
+
+        public Board(List<Piece> pieces, Color currentTurnColor)
         {
             this.Pieces = pieces;
-            Position = MapPiecesToDictionary();
+            this.currentTurnColor = currentTurnColor;
+            Positions = new Dictionary<Tile, Piece>();
+            MapPiecesToDictionary();
         }
-        public Dictionary<Tile, Piece> MapPiecesToDictionary()
+
+        public void MapPiecesToDictionary()
```

Comment:
same comment as on IBoard, this should be an internal method for the concrete class, hence this should be marked private